### PR TITLE
TG-9: Creation of edit function on Super Admins / and delete Super Admins

### PR DESCRIPTION
**Pull request by SteampunkMatu**
Added Edit function on Super Admins section

Edit: Added delete function.

### CODE REVIEW

File: src/controllers/super-admins.js [hunk 1]
```diff
@@ -1,90 +1,45 @@
...
-  const userId = parseInt(req.params.id, 10);
-  const findUser = superAdmins.find((user) => user.id === userId);
-  if (findUser) {
-    res.status(200).json({
-      data: findUser,
+const editSuperAdmins = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const result = await SuperAdmins.findByIdAndUpdate(
+      { _id: id },
```
Comment (GinaSchP) [inline src/controllers/super-admins.js:7]: this line must be SuperAdmins.findByInAndUpdate(id, req.body, { new: true },)

File: src/controllers/super-admins.js [hunk 2]
```diff
@@ -1,90 +1,45 @@
...
+    return res.status(200).json({
+      message: `Super Admin with ID ${id} edited.`,
+      data: result,
+      error: false,
     });
-  } else {
-    res.status(404).json({
-      error: 'User not found',
+  } catch (error) {
+    return res.json({
```
Comment (GinaSchP) [inline src/controllers/super-admins.js:17]: The ststus of error is missing, must be: 
return res.status(400).json({
               message: error, // o error.toString()
               data: undefined,
               error: true
           });

File: src/controllers/super-admins.js [hunk 3]
```diff
@@ -1,90 +1,45 @@
...
+      data: result,
+      error: false,
     });
-  } else {
-    fs.writeFile('src/data/super-admins.json', JSON.stringify(newSuperAdmins, null, 2), () => {
-      res.status(200).json({
-        message: 'Created new super admin',
-      });
+  } catch (error) {
+    return res.json({
```
Comment (GinaSchP) [inline src/controllers/super-admins.js:35]: same as above

**Review by GinaSchP — CHANGES_REQUESTED** [review]

File: src/index.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import express from 'express';
 import mongoose from 'mongoose';
+import superAdminsRoutes from './routes/super-admins';
```
Comment (lmalvasia) [inline src/index.js:3]: You have to import this inside the ./routes/index.js file (check the Fede's PR)

File: src/models/Super-admins.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import mongoose from 'mongoose';
+
+const { Schema } = mongoose;
+
+const superAdminsSchema = new Schema({
+
```
Comment (lmalvasia) [inline src/models/Super-admins.js:6]: remove this space

File: src/models/Super-admins.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  name: { type: String, require: true },
+  last_name: { type: String, require: true },
+  email: { type: String, require: true },
+  password: { type: String, require: true },
+
```
Comment (lmalvasia) [inline src/models/Super-admins.js:11]: remove this another space

File: src/validations/super-admins.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import Joi from 'joi';
+
+const validateCreation = (req, res, next) => {
```
Comment (lmalvasia) [inline src/validations/super-admins.js:3]:
```suggestion
const validateSuperAdminBody = (req, res, next) => {
```

Comment (lmalvasia) [reply src/validations/super-admins.js:3]: or 
```suggestion
const validateSuperAdminCreationBody = (req, res, next) => {
```

File: src/validations/super-admins.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    return res.status(400).json({
+      message: `There was an error: ${validation.error.details[0].message}`,
+      data: undefined,
+      error: true,
+    });
+  }
+  return next();
+};
+
+export default {
```
Comment (lmalvasia) [inline src/validations/super-admins.js:21]: export default superAdminValidation;

File: src/controllers/super-admins.js [hunk 9]
```diff
@@ -1,90 +1,43 @@
...
-  };
-  const newSuperAdmins = [...superAdmins, user];
-  if (!user.email || !user.first_name || !user.last_name
-          || !user.date_of_birth || !user.DNI || !user.Phone) {
-    res.status(404).json({
-      error: 'Cannot create a super admin. Invalid body params',
+const deletedSuperAdmins = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const result = await SuperAdmins.findByIdAndDelete(id);
```
Comment (lmalvasia) [inline src/controllers/super-admins.js:24]: What happen when we send an invalid id? (Super admin does not exist)

File: src/controllers/super-admins.js [hunk 10]
```diff
@@ -1,90 +1,43 @@
...
-  fs.writeFile('src/data/super-admins.json', JSON.stringify(deleteUser), (err) => {
-    if (err) {
-      res.send('Cannot deleted user');
-    } else {
-      res.status(200).json({
-        message: 'Super admin deleted',
-      });
-    }
-  });
+export default {
```
Comment (lmalvasia) [inline src/controllers/super-admins.js:40]: Remove this export default and export the functions individually

**Review by lmalvasia — APPROVED** [review]

**Review by GinaSchP — APPROVED** [review]

**Pull request merged**